### PR TITLE
list all column api properties

### LINKED ISSUE #3198: Column API: list all properties available
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/dbt-classes#column-api

### What part(s) of the page would you like to see updated?

The docs should include all the [methods decorated with `@property`](https://github.com/dbt-labs/dbt-core/blob/a88f64039583b1fce764408c841b22963d4908b8/core/dbt/adapters/base/column.py#L32-L47) as well as the explicit properties listed [here](https://github.com/dbt-labs/dbt-core/blob/a88f64039583b1fce764408c841b22963d4908b8/core/dbt/adapters/base/column.py#L17-L21).

Decorated properties:
- name
- quoted
- data_type

Explicit properties:
- column
- dtype
- char_size
- numeric_precision
- numeric_scale

The unified list is:
- column: Returns the name of the column
- name: Returns the name of the column (alias for `column`)
- quoted: Returns the name of the column wrapped in quotes
- data_type: Returns the data type of the column (with size/precision/scale included)
- dtype: Returns the data type of the column (without any size/precision/scale included)
- char_size: Returns the maximum size for character varying columns
- numeric_precision: Returns the maximum precision for fixed decimal columns
- numeric_scale: Returns the maximum scale for fixed decimal columns

### Additional information

Instigating context: in discussing options within https://github.com/dbt-labs/dbt-codegen/pull/122, there weren't sufficient docs to describe all the options and their differences.

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr adds the full properties available to the column api doc (dbt classes). 

Resolves #3198

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-add-properties-column-api-dbt-labs.vercel.app/reference/dbt-classes

### CODE REVIEW

File: website/docs/reference/dbt-classes.md [hunk 1]
```diff
@@ -98,9 +98,14 @@ col.numeric_type('numeric', 12, 4) # numeric(12,4)
 
 ### Properties
 
-- **name**: Returns the name of the column
+- **char_size**: Returns the maximum size for character varying columns
+- **column**: Returns the name of the column
+- **data_type**: Returns the data type of the column (with size/precision/scale included)
+- **dtype**: Returns the data type of the column (without any size/precision/scale included)
+- **name**: Returns the name of the column (alias for column)
```
Comment (matthewshaver) [inline website/docs/reference/dbt-classes.md]: Would it help to include language about how this differs from `column` above?

**Review by matthewshaver — APPROVED** [review]

File: website/docs/reference/dbt-classes.md (hunk 1 above)
Comment (mirnawong1) [reply website/docs/reference/dbt-classes.md]: hey @dbeatty10 can you help clarify the difference?

Comment (mirnawong1) [reply website/docs/reference/dbt-classes.md]: from looking at the issue, `name` is  another way to access the same info -- alias for column. so it's included for convenience or consistency with other naming conventions.

what about the below @matthewshaver :

name: Returns the name of the column (identical to column, provided as an alias).

Comment (mirnawong1) [inline website/docs/reference/dbt-classes.md]:
```suggestion
- **name**: Returns the name of the column (identical to `column`, provided as an alias).
```

**Pull request merged**